docs/architecture/retry.md

```diff
@@ -0,0 +1,38 @@
+# Retry Mechanisms
+
+## gRPC
+
+### Interceptor
+
+All services use the gRPC interceptor `interceptor.RetryableInterceptor` that can retry a failed gRPC request.
```

Comment:
We do both client-side and server-side retries and it's not clear which one you're talking about here. You should describe both. Actually I think client-side retries are more important, the server-side ones are fairly limited (deliberately)

Reply:
Ha! I'll describe the client-side one, too 👍